streams/src/test/java/org/apache/kafka/streams/integration/IQv2StoreIntegrationTest.java

```diff
@@ -0,0 +1,557 @@
+
+        public boolean global() {
+            return false;
+        }
+    }
+
+    @Parameterized.Parameters(name = "cache={0}, log={1}, supplier={2}")
+    public static Collection<Object[]> data() {
+        final List<Object[]> values = new ArrayList<>();
+        for (final boolean cacheEnabled : Arrays.asList(true, false)) {
+            for (final boolean logEnabled : Arrays.asList(true, false)) {
+                for (final StoresToTest toTest : StoresToTest.values()) {
+                    values.add(new Object[]{cacheEnabled, logEnabled, toTest.name()});
+                }
+            }
+        }
+        return values;
+    }
+
+    public IQv2StoreIntegrationTest(
+        final boolean cache,
+        final boolean log,
+        final String storeToTest) {
+
+        this.cache = cache;
+        this.log = log;
+        this.storeToTest = StoresToTest.valueOf(storeToTest);
+    }
+
+    @BeforeClass
+    public static void before()
+        throws InterruptedException, IOException, ExecutionException, TimeoutException {
+        CLUSTER.start();
+        final int partitions = 2;
+        CLUSTER.createTopic(INPUT_TOPIC_NAME, partitions, 1);
+
+        final Properties producerProps = new Properties();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class);
+
+        final List<Future<RecordMetadata>> futures = new LinkedList<>();
+        try (final Producer<Integer, Integer> producer = new KafkaProducer<>(producerProps)) {
+            for (int i = 0; i < 3; i++) {
+                final Future<RecordMetadata> send = producer.send(
+                    new ProducerRecord<>(
+                        INPUT_TOPIC_NAME,
+                        i % partitions,
+                        Time.SYSTEM.milliseconds(),
+                        i,
+                        i,
+                        null
+                    )
+                );
+                futures.add(send);
+                Time.SYSTEM.sleep(1L);
+            }
+            producer.flush();
+
+            for (final Future<RecordMetadata> future : futures) {
+                final RecordMetadata recordMetadata = future.get(1, TimeUnit.MINUTES);
+                assertThat(recordMetadata.hasOffset(), is(true));
+                INPUT_POSITION.withComponent(
+                    recordMetadata.topic(),
+                    recordMetadata.partition(),
+                    recordMetadata.offset()
+                );
+            }
+        }
+
+        assertThat(INPUT_POSITION, equalTo(
+            Position
+                .emptyPosition()
+                .withComponent(INPUT_TOPIC_NAME, 0, 1L)
+                .withComponent(INPUT_TOPIC_NAME, 1, 0L)
+        ));
+    }
+
+    @Before
```

Comment:
For each store configuration, we will need to run a slightly different Streams application, so before each test, we create Streams and then tear it down (and clean up) afterwards.